pkg/storj/pieceid.go

```diff
@@ -0,0 +1,109 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package storj
+
+import (
+	"crypto/hmac"
+	"crypto/rand"
+	"crypto/sha512"
+	"encoding/hex"
+
+	"github.com/zeebo/errs"
+)
+
+// ErrPieceID is used when something goes wrong with a piece ID
+var ErrPieceID = errs.Class("piece ID error")
+
+// PieceID2 is the unique identifier for pieces
+type PieceID2 [32]byte
+
+// NewPieceID creates a piece ID
+func NewPieceID() PieceID2 {
+	var id PieceID2
+
+	_, err := rand.Read(id[:])
+	if err != nil {
+		panic(err)
```

Comment:
In other places in the code base we are returning error instead of panic if `rand.Rand` fails. I know such errors should not happen, but do we really want to crash the satellite if this ever happens?

Reply:
If we have a broken random then the system has completely failed and pretty much should be set on fire.